src/lib.rs

```diff
@@ -63,52 +51,30 @@ fn get_generic_array(item: &impl OMFE, key: &str, is_decrypt: bool) -> (Key<Aes2
 
     // deferencing the Generic block from the Vec as a single Unit
     //let deref_generic_block = generic_block.as_mut_slice();
-
-    (key.to_owned(), generic_block)
-}
-
-/// I have no idea why you would need this trait for public use, but here you go
-pub trait OMFE {
-    fn get_raw_bytes(&self) -> Vec<u8>;
+    generic_block
 }
 
 /// This is my create
 #[cfg(test)]
 mod tests {
-    use crate::{decrypt::Decryptor, encrypt::Encryptor};
+    use crate::Aes256Cryptor;
 
     #[test]
     fn test_with_strings() {
         let my_32byte_key = "Thisi$MyKeyT0Encryp!thislastTime".to_owned();
         let original_text = "I am Omkaram Venkatesh and 
         this is my plain text and some random chars 223@#$^$%*%^(!#@%$~@#$[]]'///\\drewe. Lets see if this gets encrypted now)".to_string();
 
-        let encrypt_obj: Encryptor = Encryptor::from(original_text.as_str());
-        let encrypted_bytes: Vec<u8> = encrypt_obj.encrypt_with(&my_32byte_key);
+        let cryptor = Aes256Cryptor::try_from(&my_32byte_key).unwrap();
+        let encrypted_bytes: Vec<u8> = cryptor.encrypt(&original_text as &str);
 
-        let decrypted_obj: Decryptor = Decryptor::from(encrypted_bytes);
-        let decrypted_bytes: Vec<u8> = decrypted_obj
-            .decrypt_with(&my_32byte_key)
-            .unwrap_or_default();
-        let decrypted_text: String = String::from_utf8_lossy(&decrypted_bytes).to_string();
+        let decrypted_text: String =
+            String::from_utf8_lossy(&cryptor.decrypt(encrypted_bytes).unwrap_or_default())
+                .to_string();
 
         assert_eq!(original_text, decrypted_text);
     }
 
     #[test]
-    fn test_with_bytes() {
```

Comment:
@xmh0511 Everything looks nice and clean. Just this one comment. Can you bring this test back? It might look redundant but it helps to show the encrypt function also works directly on vec<u8> instead of &str. I want to assert_eq!(original_bytes, decrypted_bytes);  similar to assert_eq!(original_text, decrypted_text) in fn test_with_strings(). Let me know what you think.

Reply:
Ah, I forgot this test function, I prepared to copy the content in byte.rs to this function, which is exactly about processing the pure bytes.